src/components/accordion/accordion.tsx

```diff
@@ -367,14 +367,18 @@ export class EuiAccordionClass extends Component<
       childrenContent = children;
     }
 
+    // We want to omit the aria-expanded attribute when the second
+    // element is not a button. Otherwise axe-core throws a violation.
+    const isExpandableButton = ButtonElement === 'button';
+
     const button = (
       <ButtonElement
         css={cssButtonStyles}
         {...buttonProps}
         id={buttonId}
         className={buttonClasses}
         aria-controls={id}
-        aria-expanded={isOpen}
+        aria-expanded={isExpandableButton ? isOpen : undefined}
```

Comment:
my 2c: I don't think this needs to be pulled out to a variable. I don't think doing so adds significant readability as to keeping it inline

```suggestion
        // `aria-expanded` is only a valid attribute on interactive controls - axe-core throws a violation otherwise
        aria-expanded={ButtonElement === 'button' ? isOpen : undefined}
```

Reply:
I was so close to keeping that inline then talked myself out of it. I'll grab your commit suggestion.